test/distributed/test_manager.py

```diff
@@ -167,7 +179,42 @@ def test_manager_singleton():
     assert manager_1.group_name() == manager_2.group_name()
     assert manager_1.broadcast_buffers == manager_2.broadcast_buffers
     assert manager_1.find_unused_parameters == manager_2.find_unused_parameters
+    DistributedManager.cleanup()
+    del os.environ["RANK"]
+    del os.environ["WORLD_SIZE"]
+
+
+@distributed_test
+def test_manager_single_process_subgroups():
+    os.environ["RANK"] = "0"
+    os.environ["WORLD_SIZE"] = "1"
+    os.environ["MASTER_ADDR"] = "localhost"
+    os.environ["MASTER_PORT"] = str(12355)
+
+    # Reset class state
     DistributedManager._shared_state = {}
```

Comment:
Should this be necessary if `DistributedManager.cleanup()` is used throughout?

Reply:
No, this is not necessary. Removed.